uniffi_bindgen/src/interface/universe.rs

```diff
@@ -71,46 +71,13 @@ impl TypeUniverse {
         if !self.all_known_types.contains(type_) {
             self.all_known_types.insert(type_.to_owned());
         }
-        match type_ {
-            Type::UInt8 => self.add_type_definition("u8", type_)?,
-            Type::Int8 => self.add_type_definition("i8", type_)?,
-            Type::UInt16 => self.add_type_definition("u16", type_)?,
-            Type::Int16 => self.add_type_definition("i16", type_)?,
-            Type::UInt32 => self.add_type_definition("i32", type_)?,
-            Type::Int32 => self.add_type_definition("u32", type_)?,
-            Type::UInt64 => self.add_type_definition("u64", type_)?,
-            Type::Int64 => self.add_type_definition("i64", type_)?,
-            Type::Float32 => self.add_type_definition("f32", type_)?,
-            Type::Float64 => self.add_type_definition("f64", type_)?,
-            Type::Boolean => self.add_type_definition("bool", type_)?,
-            Type::String => self.add_type_definition("string", type_)?,
-            Type::Bytes => self.add_type_definition("bytes", type_)?,
-            Type::Timestamp => self.add_type_definition("timestamp", type_)?,
-            Type::Duration => self.add_type_definition("duration", type_)?,
-            Type::Object { name, .. }
-            | Type::Record { name, .. }
-            | Type::Enum { name, .. }
-            | Type::CallbackInterface { name, .. }
-            | Type::External { name, .. } => self.add_type_definition(name, type_)?,
-            Type::Custom { name, builtin, .. } => {
-                self.add_type_definition(name, type_)?;
-                self.add_known_type(builtin)
-                    .with_context(|| format!("adding custom builtin type {builtin:?}"))?;
-            }
-            // Structurally recursive types.
-            Type::Optional { inner_type, .. } | Type::Sequence { inner_type, .. } => {
-                self.add_known_type(inner_type)
-                    .with_context(|| format!("adding optional type {inner_type:?}"))?;
-            }
-            Type::Map {
-                key_type,
-                value_type,
-            } => {
-                self.add_known_type(key_type)
-                    .with_context(|| format!("adding map key type {key_type:?}"))?;
-                self.add_known_type(value_type)
-                    .with_context(|| format!("adding map value type {value_type:?}"))?;
-            }
+        // all sub-types, but we want to skip this type as we just added it above.
+        for sub in type_.iter_types().skip(1) {
```

Comment:
This relies on the type being the first thing generated from the iterator which could fail.  What about removing the `skip` and the `add_known_type` call above?  If that doesn't work, maybe this could be a `filter` with an equality check rather than a skip.

Reply:
> skip and the add_known_type call above?

As per the comment above, that would mean many extra clones which are currently avoided. I'm not sure that matters though - wdyt?

Reply:
> In theory, this could be a breaking change for external bindings authors, is it worth adding a CHANGELOG entry?

TBH I don't see how - IIUC, that theoretical scenario would be that they ask for a known type named "u8" just so they can get back a `Type::U8` - wouldn't they just use the type directly? Or is there something else I'm missing here?

Reply:
Ahh, I understand now.  I don't think it matters much but what about splitting out `iter_nested_types()` from `iter_types`?  That avoids the clone and seems less fragile than the `skip(1)`.

Reply:
> > In theory, this could be a breaking change for external bindings authors, is it worth adding a CHANGELOG entry?
> 
> TBH I don't see how - IIUC, that theoretical scenario would be that they ask for a known type named "u8" just so they can get back a `Type::U8` - wouldn't they just use the type directly? Or is there something else I'm missing here?

I think that would have to be the chain of events.  I'm fine not adding anything since it seems pretty unlikely, I just wanted to ask the question.